processor/src/main/java/com/linecorp/decaton/processor/HashableKey.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * This is a wrapper for record keys in {@code byte[]} to support {@link #equals(Object)} and {@link #hashCode()},
+ * so that they can be stored as a key or item of HashMap or HashSet.
+ *
+ * It also supports caching the computed hash code like {@link String}.
+ */
+public class HashableKey {
```

Comment:
- Let's move this class to under `decaton/processor/internal` as this class isn't intended to be used by users.
- Hm, `HashableKey` sounds bit too general? Despite this class is intended only to wrap byte array.
  * How about `HashableByteArray`? or better naming is welcome...

Reply:
https://github.com/line/decaton/pull/157/commits/9d24b494083f52bc7b19578009fd1476770ac12d

> Let's move this class to under decaton/processor/internal

Moved to `decaton.processor.runtime.internal` (there's no `decaton.processor.internal` so you meant this package?).

> Hm, HashableKey sounds too general?

Renamed the class to `HashableByteArray` along with its fields.